### DISCUSSION TITLE
Add an option to close the Off-Canvas widget when a link inside is clicked

### DISCUSSION THREAD

**Discussion by cmasetti** [elementor/elementor#27727] [Feature Request]
### Prerequisites

- [X] I have searched for similar feature requests in both open and closed discussions and cannot find an existing request.
- [X] I have verified that the feature is still missing from the latest stable versions of Elementor or Elementor Pro.

### What problem is your feature request going to solve? Please describe..

I installed the new Offcanvas widget to my header (https://toptable.blackt.agency/)
It works fine for page links. But one of my menu items is an anchor link ("restaurants") to a section of the homepage.
When I click on "restaurants" it scrolls to the right section. But if I am in the home already, the offcanvas does not close after the click, it stays open.

The following code works for popups, if we could only update it to Offcanvas it would solve the issue.
<script>
jQuery(function($){
$(document).on('click','.elementor-location-popup a', function(event){
elementorProFrontend.modules.popup.closePopup( {}, event);
});
});
</script>

### Describe the solution you'd like.

Adapt the following code to work for Offcanvas (instead of Popups as it is now)
<script>
jQuery(function($){
$(document).on('click','.elementor-location-popup a', function(event){
elementorProFrontend.modules.popup.closePopup( {}, event);
});
});
</script>

### Agreement

- [X] I agree that my request may be closed without action if it doesn't meet all the requirements.

**Comment by Vossha** [comment]
+1

I brought this up in the beta version discussion. So I'll copy the feedback I gave there:

> Would love if the off-canvas modal automatically closes once an anchor link in the off-canvas container is clicked.
> 
> I have multiple use cases where I would like to include the Table Of Contents in the off-canvas modal but once a link in the TOC is clicked, I want the modal to close. Or, for 1-page websites with navigation menus consisting only of anchor links, it would be nice for the user experience if the modal closes automatically.
> 
> https://github.com/orgs/elementor/discussions/27371#discussioncomment-9496323

**Comment by SMolodoy** [comment]
+1

**Comment by arnasofc** [comment]
+1

**Comment by TomRaraAvis** [comment]
+1
#27727 — Add an option to close the Off-Canvas widget when a link inside is clicked
#8125 — Trigger to close popup
#7085 — Popup - Close when you click on a button
#28853 — hiding offcanvas doesn't work well with 2 or more. / problem shown in the video
#16977 — Popup close on any link/button click
#29593 — Close Off-Canvas with any Button Located within the Off-Canvas "popup"

> @TomRaraAvis There is a Dynamic Tag to Close the Off-Canvas. The Dynamic Tag is available on anything that has a link (Button, heading, icon etc).

yes, but i need the link to scroll to an anchor & also close the off-canvas — double duty. solution?

<img width="213" alt="image" src="https://github.com/user-attachments/assets/de3fac78-9b3f-48d1-843a-d9f5ed55e064" />
